test/RibbonThetaVault.ts

```diff
@@ -947,6 +1110,118 @@ function behavesLikeRibbonOptionsVault(params: {
       });
     });
 
+    if (isUsdcAuction) {
+      describe("#setSwapPath", () => {
+        time.revertToSnapshotAfterTest();
+
+        it("reverts when path is too short", async function () {
+          await expect(
+            vault.connect(ownerSigner).setSwapPath("0x")
+          ).to.be.revertedWith("Path too short");
+        });
+
+        it("reverts when path is too long", async function () {
+          await expect(
+            vault
+              .connect(ownerSigner)
+              .setSwapPath(
+                encodePath(
+                  [
+                    USDC_ADDRESS[chainId],
+                    WETH_ADDRESS[chainId],
+                    USDC_ADDRESS[chainId],
+                    WETH_ADDRESS[chainId],
+                    asset,
+                  ],
+                  [10000, 10000, 10000, 10000]
+                )
+              )
+          ).to.be.revertedWith("Path too long");
+        });
+
+        it("reverts when pool does not exist", async function () {
+          let ref: string;
+
+          dexFactory = await getContractAt(
+            "IUniswapV3Factory", // Supports Uniswap V3 only
+            DEX_FACTORY[chainId]
+          );
+
+          let pool = await dexFactory.getPool(asset, asset, 10000);
+          if (pool == constants.AddressZero) {
```

Comment:
Isn't `pool` guaranteed to be `address(0)`? Since you are doing `getPool(asset, asset)` which does not exist

Reply:
Yes, this is to prove that the `dexFactory` indeed returns a zero address. And therefore matches the result we are getting from `setSwapPath`.